lib-ce/emqx_modules/src/emqx_mod_rewrite.erl

```diff
@@ -75,30 +74,60 @@ description() ->
 %% Internal functions
 %%--------------------------------------------------------------------
 
-log_start(Rules) ->
-    PubRules = [{pub, Topic, Re, Dest} || {rewrite, pub, Topic, Re, Dest} <- Rules],
-    SubRules = [{sub, Topic, Re, Dest} || {rewrite, sub, Topic, Re, Dest} <- Rules],
+compile_rules(RawRules) ->
+    compile(validate_rules(RawRules)).
+
+compile({PubRules, SubRules}) ->
+    CompileRE =
+        fun({rewrite, Source, Re, Dest}) ->
+                {ok, MP} = re:compile(Re),
+                {rewrite, Source, MP, Dest}
+        end,
+    {lists:map(CompileRE, PubRules), lists:map(CompileRE, SubRules)}.
+
+validate_rules(Rules) ->
+    validate_rules(Rules, {[], []}).
+
+validate_rules([], {PubRules, SubRules}) ->
     ?LOG(info, "[Rewrite] Load: pub rules count ~p sub rules count ~p",
         [erlang:length(PubRules), erlang:length(SubRules)]),
-    log_rule(PubRules, 1),
-    log_rule(SubRules, 1).
+    log_rules(pub, PubRules),
+    log_rules(sub, SubRules),
+    {PubRules, SubRules};
```

Comment:
Double check: should we reverse or NOT ?
if not, add a comment here to explain why NOT.
```suggestion
    {lists:reverse(PubRules), lists:reverse(SubRules)};
```